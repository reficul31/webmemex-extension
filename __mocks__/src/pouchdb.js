import PouchDB from 'pouchdb-node'
import PouchDBMemory from 'pouchdb-adapter-memory'

PouchDB.plugin(PouchDBMemory)

const pouchdbOptions = {
    name: 'testdb',
    auto_compaction: true,
    adapter: 'memory',
}

const db = PouchDB(pouchdbOptions)
export default db
