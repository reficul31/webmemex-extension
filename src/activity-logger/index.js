// Stuff that is to be accessible from other modules (folders)

import docuri from 'docuri'
import randomString from 'src/util/random-string'

export const visitKeyPrefix = 'visit/'

/**
* Creates an _id string given the variables, or vice versa parses such strings
* We simply use the creation time for the id, for easy chronological sorting.
* We add a random string we call a 'nonce' to prevent accidental collisions.
*
* @param {Number} timestamp - timestamp of the document
* @param {Number} nonce - random 10 digit number
*
* @returns {String} - string id of the document
*/
export const convertVisitDocId = docuri.route(`${visitKeyPrefix}:timestamp/:nonce`)

/**
* Extract the type, timestamp and nonce from an ID of a document
*
* @param {String} - string id of the document
*/
const convertAnyTimestampedDocId = docuri.route(':type/:timestamp/:nonce')
export const getTimestamp = doc =>
    Number.parseInt(convertAnyTimestampedDocId(doc._id).timestamp)

/**
* Generate a visit document ID from the timestamp and a random string
*
* @param {Number} timestamp - timestamp of the visit document
* @param {Number} nonce - random 10 digit number
*
* @returns {string} - document id of the visit document
*/
export function generateVisitDocId({timestamp, nonce} = {}) {
    const date = timestamp ? new Date(timestamp) : new Date()
    return convertVisitDocId({
        timestamp: date.getTime(),
        nonce: nonce || randomString(),
    })
}

/**
* Tell whether a page can be stored
*
* @param {string} url - url of the visit document
*
* @returns {boolean} - whether or not the page is loggable
*/
export function isLoggable({url}) {
    // Only http(s) pages. Ignoring data URLs, newtab, ...
    const loggableUrlPattern = /^https?:\/\//
    return loggableUrlPattern.test(url)
}

/**
* Tell whether a page should be stored
*
* @param {string} url - url of the visit document
*
* @returns {boolean} - whether or not the page should be stored
*/
export function shouldBeLogged({url}) {
    // Currently, we log everything that we think we can log.
    // TODO Add ignore-/blacklist lookup here (issue #22, #94)
    return isLoggable({url})
}
