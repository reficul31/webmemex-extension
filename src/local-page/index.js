import get from 'lodash/fp/get'

/**
* Extract the local page URL from the page instance.
*
* @param {Page} page - stored page in the database
*
* @returns {string} - URL used to refer to the local page
*/
export function hrefForLocalPage({page}) {
    // Check if it has a stored page attached at all.
    if (!get(['_attachments', 'frozen-page.html'])(page)) {
        return undefined
    }
    const pageId = page._id
    const url = `/local-page.html?page=${encodeURIComponent(pageId)}`
    const hash = (page.url && page.url.split('#')[1])
    const href = (hash !== undefined) ? url + '#' + hash : url
    return href
}
