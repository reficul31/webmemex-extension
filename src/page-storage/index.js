import docuri from 'docuri'
import randomString from 'src/util/random-string'

export const pageKeyPrefix = 'page/'

export const convertPageDocId = docuri.route(`${pageKeyPrefix}:timestamp/:nonce`)

/**
* Generate a page doc id
*
* @param {Number} timestamp - timestamp of the page
* @param {string} nonce - random string of 10 digit number
*
* @returns {string} - doc id of the page as a string
*/
export function generatePageDocId({timestamp, nonce} = {}) {
    const date = timestamp ? new Date(timestamp) : new Date()
    return convertPageDocId({
        timestamp: date.getTime(),
        nonce: nonce || randomString(),
    })
}
