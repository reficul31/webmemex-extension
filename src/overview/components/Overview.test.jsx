/* eslint-env jest */
/* eslint import/namespace: "off" */

import React from 'react'
import Overview from 'src/overview/components/Overview'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import * as db from 'src/pouchdb'

import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)

function setupStore(overrides) {
    const overview = Object.assign({
        searchResult: {rows: []},
        currentQueryParams: {
            query: '',
            startDate: undefined,
            endDate: undefined,
        },
        activeQueryParams: undefined,
        waitingForResults: false,
    }, overrides)
    return mockStore({overview})
}

describe('Overview', () => {
    beforeAll(() => {
        db.getAttachmentAsDataUri = jest.fn().mockReturnValue(
            new Promise((resolve, reject) => {
                resolve()
            })
        )
    })

    test('should test against the given snapshot', () => {
        const store = setupStore()
        const component = shallow(
            <Overview store={store} />
        )
        expect(component).toMatchSnapshot()
    })

    test('should setQuery when input is changed', () => {
        expect.assertions(2)
        const store = setupStore()
        const component = mount(
            <Provider store={store}>
                <Overview />
            </Provider>
        )
        const queryInput = component.find('.queryInputComponent').find('input')
        queryInput.simulate('change', { target: {value: 'test'} })
        expect(store.getActions()[0].type).toBe('[2] overview/setQuery')
        expect(store.getActions()[0].payload).toBe('test')
    })

    test('should clear the query when escape is pressed', () => {
        expect.assertions(2)
        const store = setupStore()
        const component = mount(
            <Provider store={store}>
                <Overview />
            </Provider>
        )
        const queryInput = component.find('.queryInputComponent').find('input')
        queryInput.simulate('focus')
        queryInput.simulate('keyDown', {key: 'Escape'})
        expect(store.getActions()[0].type).toBe('[2] overview/setQuery')
        expect(store.getActions()[0].payload).toBe('')
    })

    test('should set endDate when input is changed', () => {
        expect.assertions(2)
        let expected = expect.stringMatching('^[0-9]{13}$')
        const store = setupStore()
        const component = mount(
            <Provider store={store}>
                <Overview />
            </Provider>
        )
        const dateInput = component.find('DateSelection').find('input')
        dateInput.simulate('change', {target: {value: '02-03-2017'}})
        expect(store.getActions()[0].type).toBe('[4] overview/setEndDate')
        expect(store.getActions()[0].payload).toEqual(expected)
    })

    test('should load more results when bottom is reached', () => {
        const store = setupStore()
        const component = mount(
            <Provider store={store}>
                <Overview />
            </Provider>
        )
        component.find('ResultList').prop('onBottomReached')()
        expect(store.getActions()[0].type).toBe('[11] pending')
    })
})
