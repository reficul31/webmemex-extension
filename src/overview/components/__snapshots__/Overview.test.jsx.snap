// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should test against the given snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Overview
    endDate={undefined}
    onBottomReached={[Function]}
    onEndDateChange={[Function]}
    onInputChanged={[Function]}
    query=""
    searchResult={
        Object {
            "rows": Array [],
          }
    }
    startDate={undefined}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": null,
          }
    }
    waitingForResults={false}
/>,
  "nodes": Array [
    <Overview
      endDate={undefined}
      onBottomReached={[Function]}
      onEndDateChange={[Function]}
      onInputChanged={[Function]}
      query=""
      searchResult={
            Object {
                  "rows": Array [],
                }
      }
      startDate={undefined}
      store={
            Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                }
      }
      storeSubscription={
            Subscription {
                  "listeners": Object {
                    "notify": [Function],
                  },
                  "onStateChange": [Function],
                  "parentSub": undefined,
                  "store": Object {
                    "clearActions": [Function],
                    "dispatch": [Function],
                    "getActions": [Function],
                    "getState": [Function],
                    "replaceReducer": [Function],
                    "subscribe": [Function],
                  },
                  "unsubscribe": null,
                }
      }
      waitingForResults={false}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(Overview)
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "notifyNestedSubs": [Function],
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "endDate": undefined,
            "onBottomReached": [Function],
            "onEndDateChange": [Function],
            "onInputChanged": [Function],
            "query": "",
            "searchResult": Object {
              "rows": Array [],
            },
            "startDate": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "waitingForResults": false,
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 1,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Overview
          endDate={undefined}
          onBottomReached={[Function]}
          onEndDateChange={[Function]}
          onInputChanged={[Function]}
          query=""
          searchResult={
                    Object {
                              "rows": Array [],
                            }
          }
          startDate={undefined}
          store={
                    Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "clearActions": [Function],
                                "dispatch": [Function],
                                "getActions": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": null,
                            }
          }
          waitingForResults={false}
/>,
        "_debugID": 2,
        "_renderedOutput": <Overview
          endDate={undefined}
          onBottomReached={[Function]}
          onEndDateChange={[Function]}
          onInputChanged={[Function]}
          query=""
          searchResult={
                    Object {
                              "rows": Array [],
                            }
          }
          startDate={undefined}
          store={
                    Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "clearActions": [Function],
                                "dispatch": [Function],
                                "getActions": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": null,
                            }
          }
          waitingForResults={false}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(Overview)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
}
`;
